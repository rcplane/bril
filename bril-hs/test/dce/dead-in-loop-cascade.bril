@main {
  i: int = const 0;
  j: int = const 0;
  one: int = const 1;
  five: int = const 5;
.loop.header:
  done: bool = eq i five;
  br done .loop.exit .loop.body;
.loop.body:
  i: int = add i one;
  # j is dead because k is dead
  j: int = add j one;
  # k is dead 
  k: int = id j;
  jmp .loop.header;
.loop.exit:
}