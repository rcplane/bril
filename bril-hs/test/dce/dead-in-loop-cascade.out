@main {
  i: int = const 0;
  one: int = const 1;
  five: int = const 5;
.loop.header:
  done: bool = eq i five;
  br done .loop.exit .loop.body;
.loop.body:
  i: int = add i one;
  jmp .loop.header;
.loop.exit:
}
